graylog2-server/src/main/java/org/graylog2/rest/resources/users/UsersResource.java

```diff
@@ -427,18 +452,46 @@ private UserSummary toUserResponse(org.graylog2.plugin.database.users.User user)
     }
 
     private UserSummary toUserResponse(org.graylog2.plugin.database.users.User user, boolean includePermissions) {
+
+        final List<String> permissions;
+
+        final Set<String> roleIds = user.getRoleIds();
+        Set<String> roleNames = Collections.emptySet();
+        if (!roleIds.isEmpty()) {
+            try {
+                final Map<String, Role> idMap = roleService.loadAllIdMap();
+                roleNames = Sets.newHashSet(Collections2.transform(roleIds, new Role.RoleIdToNameFunction(idMap)));
+            } catch (org.graylog2.database.NotFoundException e) {
+                LOG.error("Unable to load roles", e);
```

Comment:
Shouldn't this lead to a failing request instead of just logging the issue? Otherwise the response payload is simply wrong if the user's roles cannot be loaded for some reason.